svg.js.d.ts

```diff
@@ -583,72 +1421,62 @@ declare module "@svgdotjs/svg.js" {
-        y: number;
-        constructor();
-        constructor(position: ArrayPoint);
-        constructor(point: Point);
-        constructor(position: { x: number, y: number });
-        constructor(x: number, y: number);
-        clone(): Point;
-        morph(point: Point): this;
-        at(pos: number): Point;
-        native(): SVGPoint;
-        transform(matrix: Matrix): Point;
-    }
-
-    // pointArray.js
-    class PointArray extends SVGArray {
-        constructor();
-        constructor(points: PointArrayAlias);
-        toString(): string;
-        toLine(): {
-            x1: number;
-            y1: number;
-            x2: number;
-            y2: number;
-        };
-        parse(points: PointArrayAlias): ArrayPoint[];
-        parse(array: ArrayAlias): never;
-        move(x: number, y: number): this;
-        size(width?: number, height?: number): this;
-        bbox(): Box;
-    }
-
     // poly.js
     interface poly {
+        /**
+         * DO NOT USE THIS, it for test purpose only
+         */
+        _array: PointArray
+
         array(): PointArray;
+        plot(): PointArray
         plot(p: PointArrayAlias): this;
-        move(x: number, y: number): this;
-        size(width: number, height: number): this;
+        clear(): this;
+        move(x: NumberAlias, y: NumberAlias): this;
+        size(width: number, height?: number): this;
+    }
+
+    // pointed.js
+    interface pointed {
+        x(): number
+        x(x: number): this
+        y(): number
+        y(y: number): this
+        height(): number
+        height(h: number): this
+        width(): number
+        width(w: number): this
     }
-    class PolyLine extends Shape implements poly {
+
+    class Polyline extends Shape implements poly, pointed {
+        node: SVGPolylineElement;
+        _array: PointArray
+
         array(): PointArray;
+        plot(): PointArray
         plot(p: PointArrayAlias): this;
         move(x: number, y: number): this;
-        size(width: number, height: number): this;
+        size(width: number, height?: number): this;
         marker(position: string, width?: number, height?: number, block?: (marker: Marker) => void): Marker;
         marker(position: string, marker: Marker): Marker;
     }
     class Polygon extends Shape implements poly {
+        node: SVGPolygonElement;
+        _array: PointArray
         array(): PointArray;
+        plot(): PointArray
         plot(p: PointArrayAlias): this;
         move(x: number, y: number): this;
-        size(width: number, height: number): this;
+        size(width: number, height?: number): this;
         marker(position: string, width?: number, height?: number, block?: (marker: Marker) => void): Marker;
         marker(position: string, marker: Marker): Marker;
     }
     // rect.js
     class Rect extends Shape {
-        radius(x: number, y?: number): this;
-    }
+        constructor(val: { width: number, height: number, x: number, y: number })

```

Comment:
I am not sure about this one. Every constructor can take an object which is passed directly to `attr`. So this is not limmited to width, height...

Alternatively you can pass in a node into the Rect constructor.

This both statements are true for all elements in svg.js. So you can do the following:

```js
new Rect(SVGRectNode)
new Rect({width: 100, fill: 'none'})
```